client/client.go

```diff
@@ -413,6 +414,9 @@ func (cli *Client) SetCustomHTTPHeaders(headers map[string]string) {
 func (cli *Client) Dialer() func(context.Context) (net.Conn, error) {
 	return func(ctx context.Context) (net.Conn, error) {
 		if transport, ok := cli.client.Transport.(*http.Transport); ok {
+			if transport.TLSClientConfig != nil {
+				return tls.Dial(cli.proto, cli.addr, transport.TLSClientConfig)
```

Comment:
Instead of calling `Dial` directly here, could you just call https://golang.org/pkg/crypto/tls/#Client using the `net.Conn` returned from `transport.DialContext`? It doesn't seem ideal to ignore the passed in dialer and context when a tls config exists.

Reply:
Can't use Client as that would require managing the `ServerName` & `Handshake` myself but simplified to use fallback so there is only one place calling `tls.Dial`